src/core/currency/wallet/currency-wallet-pixie.ts

```diff
@@ -236,7 +234,46 @@ export const walletPixie: TamePixie<CurrencyWalletProps> = combinePixies({
             startupPromise = Promise.resolve()
               .then(() => engine.startEngine())
               .catch(error => input.props.onError(error))
+          }
+        },
+
+        destroy() {
+          const { log, walletId, walletOutput } = input.props
+          if (walletOutput == null) return
+          const { engine } = walletOutput
+
+          if (engine != null && startupPromise != null) {
+            log(`${walletId} killEngine`)
+
+            // Wait for `startEngine` to finish if that is still going:
+            startupPromise
+              .then(() => engine.killEngine())
+              .catch(error => input.props.onError(error))
+              .then(() =>
+                input.props.dispatch({
+                  type: 'CURRENCY_ENGINE_STOPPED',
+                  payload: { walletId }
+                })
+              )
+              .catch(() => {})
+          }
+        }
+      }
+    },
+    props =>
+      props.state.paused || props.walletState.paused ? undefined : props
+  ),
 
+  syncNetwork: filterPixie(
+    (input: CurrencyWalletInput) => {
+      let syncNetworkTask: PeriodicTask
+
+      return {
+        update() {
+          const { currencyInfo, walletInfo, publicWalletInfo } =
+            input.props.walletState
+          const { engine } = input.props.walletOutput
+          if (engine != null) {
```

Comment:
Before, we waited for `startEngine` to resolve before we started the `syncNetwork` task. Now we *might* call `syncNetwork` before `syncNetwork` has finished (it's racey).

We could fix the race by having `engineStarted` call `input.props.onOutput(boolean)` to indicate that the `startEngine` has finished, and we're ready to sync. Then have the new `syncNetwork` pixie wait for `input.props.walletOutput.engineStarted` to be true.